### PR TITLE
documentation for Repository, RemoteDataSource, ....

### PR DESCRIPTION
**Pull request by violoncelloCH**

### CODE REVIEW

File: app/src/main/java/com/github/swent/echo/data/repository/Repository.kt [hunk 1]
```diff
@@ -63,6 +130,12 @@ interface Repository {
      */
     suspend fun getSubTags(tagId: String): List<Tag>
 
+    /**
+     * Get all tags.
+     *
+     * @return All the existing tags.
```
Comment (unglazedstamp) [inline app/src/main/java/com/github/swent/echo/data/repository/Repository.kt:136]: this line is duplicated

File: app/src/main/java/com/github/swent/echo/data/repository/Repository.kt [hunk 2]
```diff
@@ -85,11 +172,31 @@ interface Repository {
...
+     * authenticated user.
+     *
+     * @param userId, the id of the user for which to set the profile picture
+     * @throws RepositoryStoreWhileNoInternetException in case of no internet or a network issue
+     */
     suspend fun deleteUserProfilePicture(userId: String)
 }
 
+/**
+ * Exception class in case a store operation fails due to netork conditions.
```
Comment (unglazedstamp) [inline app/src/main/java/com/github/swent/echo/data/repository/Repository.kt:196]: typo: network

**Review by unglazedstamp — CHANGES_REQUESTED** [review]
Nice work, thanks for all these comments which make the code clearer. I just found some typo, see below.

File: app/src/main/java/com/github/swent/echo/data/supabase/entities/AssociationSupabase.kt [hunk 3]
```diff
@@ -4,6 +4,15 @@ import com.github.swent.echo.data.model.Association
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
+/**
+ * Annotated version of the Association data class. Used for supabase queries.
+ *
+ * @property associationId the unique id of the association
+ * @property name the name of the association
+ * @property description the descirbtion of the association
```
[the remark this answers is not in the document]
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/data/supabase/entities/AssociationSupabase.kt]: thanks, fixed!

File: app/src/main/java/com/github/swent/echo/data/repository/Repository.kt (hunk 1 above)
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/data/repository/Repository.kt:136]: thanks, fixed

File: app/src/main/java/com/github/swent/echo/data/repository/Repository.kt (hunk 2 above)
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/data/repository/Repository.kt]: thanks, fixed

File: app/src/main/java/com/github/swent/echo/data/repository/datasources/RemoteDataSource.kt [hunk 4]
```diff
@@ -5,93 +5,303 @@ import com.github.swent.echo.data.model.Event
...
+     * @param userId The unique identifier of the target user profile.
+     * @param maxRetriesCount The maximum number of times to retry the operation if it fails.
+     *   Defaults to `RETRY_MAX`.
+     * @return An optional instance of `UserProfile`, encapsulating the queried records or null if
+     *   none was found.
+     */
     suspend fun getUserProfile(userId: String, maxRetriesCount: UInt = RETRY_MAX): UserProfile?
 
+    /**
+     * Updates the proviede user profile.
```
[the remark this answers is not in the document]
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/data/repository/datasources/RemoteDataSource.kt]: thanks, fixed

**Review by unglazedstamp — APPROVED** [review]
LGTM

**Pull request merged**